### ISSUE TITLE
[Task][Go SDK]: Update Github action for Go unit testing to use go 1.19 -> blocked on gofmt changes.

### ISSUE THREAD

**Issue by lostluck** [apache/beam#24190]
### What needs to happen?

Updating .github/workflows/go_tests.yml to use go 1.19 caused the action to fail because we have a `go fmt` check to ensure that all Go code in the repo is using the standard go formatting.

With Go 1.19, doc comments can now have richer formating, which lead to certain ad-hoc formats to become invalid, or change under `gofmt`. See https://tip.golang.org/doc/go1.19#go-doc and https://tip.golang.org/doc/comment for details on this.

While not urgent now, if we update the min SDK version to go 1.19, this becomes a minor blocker. In that event, we should disable the fmt check to allow the action to continue to succeed.

Preferably, we fix the formatting changes prior to that, and run gofmt on the repo and handle the differences, validating the rendering of content with the `godoc` tool.

It is a separate issue to take advantage of the richer documentation features of go1.19.

### Issue Priority

Priority: 3

### Issue Component

Component: sdk-go
